foodsaving/conversations/api.py

```diff
@@ -101,6 +115,52 @@ class ConversationMessageViewSet(
     def get_queryset(self):
         return self.queryset.filter(conversation__participants=self.request.user)
 
+    @detail_route(
+        methods=('POST',),
+        filter_fields=('name',),
+        permission_classes=(IsAuthenticated, IsMessageConversationParticipant)
+    )
+    def reactions(self, request, pk):
+        """route for POST /messages/{id}/reactions/ with body {"name":"emoji_name"}"""
+
+        message = get_object_or_404(ConversationMessage.objects, id=pk)
+
+        # object permissions check has to be triggered manually
+        self.check_object_permissions(self.request, message)
+
+        data = {
+            'message': pk,
+            'name': request.data['name'],
+            'user': request.user.id,
+        }
+
+        serializer = ConversationMessageReactionSerializer(data=data)
+        serializer.is_valid(raise_exception=True)
+        serializer.save()
+        return Response(serializer.data, status=status.HTTP_201_CREATED)
+
+    @detail_route(
+        methods=('DELETE',),
+        url_path='reactions/(?P<name>[a-z0-9_+-]+)',
```

Comment:
Can you add the trailing slash here, as mentioned in https://github.com/yunity/karrot-frontend/pull/926#discussion_r172047693 ?

Reply:
I overlooked this.
However when I try to add trailing slash here, the tests (which contain the trailing slash) start failing. When I put 2 trailing slashes into tests, they pass again.
So I believe the trailing slash is taken care of on a different level.
Just observation. Don't know why that happens. Perhaps [this](http://www.django-rest-framework.org/api-guide/routers/#defaultrouter)?